enterprise/views.py

```diff
@@ -538,23 +538,20 @@ def get_final_price(self, mode, request):
         """
         Get course mode's SKU discounted price after applying any entitlement available for this user.
         """
-        price = mode['min_price']
-        username = request.user.username
         try:
             client = ecommerce_api_client(request.user)
-            endpoint = getattr(
-                client.enterprise.user,
-                '{username}/products/{sku}/price'.format(sku=mode['sku'], username=username)
-            )
-            price_details = endpoint.get()
-            price = price_details.get('final_price', price)
+            endpoint = client.baskets.calculate
+            price_details = endpoint.get(sku=[mode['sku']])
+            price = price_details['total_incl_tax']
+            if price != mode['min_price']:
+                return '${}'.format(price)
         except HttpClientError:
             logger.error(
                 "Failed to get price details for course mode's SKU '{sku}' for username '{username}'".format(
-                    sku=mode['sku'], username=username
+                    sku=mode['sku'], username=request.user.username
                 )
             )
-        return '${price}'.format(price=price)
+        return mode['original_price']
```

Comment:
@itsjeyd `min_price` is the price associated with the course mode and it's an integer. It gets formatted [here](https://github.com/edx/edx-enterprise/blob/master/enterprise/views.py#L548) and then [called `original_price`](https://github.com/edx/edx-enterprise/blob/master/enterprise/views.py#L548). It looks like `$500`.

I just return `original_price` now because the template handles the discount [formatting like this](https://github.com/edx/edx-enterprise/blob/master/enterprise/templates/enterprise/enterprise_course_enrollment_page.html#L67) and because the `basket.calculate` ecommerce endpoint returns a decimal. If no discount is applied the old `${}.format(...)` produced a string like `$500.0` making the template to "think" it's a different price.